### PR TITLE
move text analysis from type_deductor to data_analyzer

### PR DESCRIPTION
**Pull request by maximlopin**
text analysis is now in `DataAnalyzer` and stored in stats_v2

### CODE REVIEW

File: mindsdb_native/libs/phases/data_analyzer/data_analyzer.py [hunk 1]
```diff
@@ -281,6 +283,14 @@ def run(self, input_data):
                 stats_v2[col_name]['nr_warnings'] += 1
             self.log.info(f'Finished analyzing column: {col_name} !\n')
```
Comment (George3d6) [inline mindsdb_native/libs/phases/data_analyzer/data_analyzer.py]: This should come after the final `if`

Comment (George3d6) [reply mindsdb_native/libs/phases/data_analyzer/data_analyzer.py]: Actually, the whole of adding the text analysis to stats-v2 should come before the above loop that logs the warnings.

File: mindsdb_native/libs/helpers/text_helpers.py [hunk 2]
```diff
@@ -10,9 +10,38 @@
...
+import flair
+
+
+def analyze_sentences(data):
+    """
+    :param data: list of str
+    :returns: 
+    tuple(
+        int: nr words total,
+        tuple(
```
Comment (George3d6) [inline mindsdb_native/libs/helpers/text_helpers.py]: this is not a tuple (anymore), it's just 3 different parameters, feel free not to leave comments for everything, is a function is small we can read it, incorrect comments do more harm than no comments.

File: mindsdb_native/libs/helpers/text_helpers.py [hunk 3]
```diff
@@ -10,9 +10,38 @@
...
+            dict: word_dist
+            dict: nr_words_dist
+        )
+    )
+    """
+    nr_words = 0
+    word_dist = Counter()
```
Comment (George3d6) [inline mindsdb_native/libs/helpers/text_helpers.py]: Why make this a `Counter ?

File: mindsdb_native/libs/helpers/text_helpers.py [hunk 4]
```diff
@@ -10,9 +10,38 @@
...
+    nr_words_dist = Counter()
+    for text in data:
+        sent = flair.data.Sentence(str(text))
+        nr_words_dist[len(sent)] += 1
+        nr_words += len(sent)
+        for tok in sent:
+            word = tok.text.strip(string.punctuation + '"\'«»')
+            word_dist[word] += 1
+
+    return nr_words, word_dist, nr_words_dist
```
Comment (George3d6) [inline mindsdb_native/libs/helpers/text_helpers.py]: [Optional]

Can probably skip adding to `nr_words` and instead just do something like:

`nr_words = sum([k*v for k,v in nr_words_dist.items()])`

**Review by George3d6 — CHANGES_REQUESTED** [review]

File: mindsdb_native/libs/helpers/text_helpers.py (hunk 3 above)
Comment (maximlopin) [reply mindsdb_native/libs/helpers/text_helpers.py]: should be `defaultdict(lambda: 0),` fixed

**Comment by maximlopin** [comment]
`tests\unit_tests\libs\phases\type_deductor\test_type_deductor.py` fails:
```
assert stats_v2[col_name]['typing']['data_type'] == expected_type
E           AssertionError: assert 'Categorical' == 'Text'
```

_update:
fixed, local tests passed_

**Review by btseytlin — APPROVED** [review]

**Pull request merged**